### PR TITLE
33/refactorAutomation

### PR DESCRIPTION
**Pull request by victorggonzalez**
This PR contains the contains the necessary workflows to automate the continuous integration and delivery of the code from this repository.

### CODE REVIEW

File: .github/workflows/cintegration-s3-caller.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ on:
 jobs:
   graasp-deploy-s3-workflow:
     name: Graasp auth
-    uses: graasp/graasp-deploy/.github/workflows/cintegration-s3.yml@6fb866ebf9c7369f4c324a3ba64e079ad54f2b49
+    uses: graasp/graasp-deploy/.github/workflows/cintegration-s3.yml@5ad156f86b2de177a81546db2ddc252a2ace36ca
```
Comment (pyphilia) [inline .github/workflows/cintegration-s3-caller.yml:19]: Actually I'm wondering what this refers to? Is it the commit hash?

**Review by pyphilia — CHANGES_REQUESTED** [review]
🚲

File: .github/workflows/cintegration-s3-caller.yml (hunk 1 above)
Comment (victorggonzalez) [reply .github/workflows/cintegration-s3-caller.yml:19]: It is the commit hash from the previous merge. Its safer to specify this as it points to an specific version of the reusable workflow that is used. So if there are changes on the main branch, this points to the same file.

File: .github/workflows/cintegration-s3-caller.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: CI pipeline for Graasp auth
+name: Continuous Integration pipeline for Graasp auth
```
[the remark this answers is not in the document]
Comment (victorggonzalez) [reply .github/workflows/cintegration-s3-caller.yml:1]: `Deploy Graasp auth to development environment`

Let me know what you think of this name

File: .github/workflows/update-staging-version.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        TAG=$(git describe --tags $COMMIT)
+        REPOSITORY=$(echo '${{ github.repository }}')
+        JSON=$(jq --null-input --arg repository "$REPOSITORY" --arg tag "$TAG" '{"repository": $repository, "tag": $tag}')
+        echo ::set-output name=json::${JSON//'%'/'%25'}
+
+    # trigger an 'on: repository_dispatch' workflow to run in graasp-deploy repository
+    - name: Repository Dispatch
+      uses: peter-evans/repository-dispatch@v2
+      with:
+        token: ${{ secrets.REPO_ACCESS_TOKEN }}
```
[the remark this answers is not in the document]
Comment (pyphilia) [reply .github/workflows/update-staging-version.yml:40]: I see, but does that mean the current value is attached to you? Thus if we remove you the permissions over the repo (we won't do that^^) the workflow will break? Or maybe this token is handled by github and you didn't save it yourself?

**Review by juancarlosfarah — APPROVED** [review]
🚀

**Pull request merged**